src/libstore/serve-protocol-impl.hh

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "serialise.hh"
+
+#include "store-api.hh"
+#include "serve-protocol.hh"
+#include "worker-protocol.hh"
+
+namespace nix {
+namespace serve_proto {
+
+struct BasicConnection
+{
+    FdSink to;
+    FdSource from;
+    int remoteVersion;
+};
+
+StorePathSet queryValidPaths(
```

Comment:
Shouldn’t this be made a method of `BasicConnection` rather than a standalone function?

Reply:
Sure. I put in a namespace like the `worker_proto` one we used earlier. Since this is sort of the "raw, secret" stuff, something about making it bare functions felt stylistically appropriate, but doesn't matter.

Reply:
Eh, I sort of feel in an "OOP is not actually good" way that it is best to just make methods functions which are intrinsically about the data type itself, and this feels to me more like something one does with the data type.

Let me know what you want @thufschmitt. If you still disagree I'll change it, but I would like this to not drag on, so Hydra can keep on being converted.